File: frontend/src/services/supplierPurchaseOrderService.js
import api from './api';

/**
 * Service Layer - Handles API calls related to supplier purchase orders.
 */

/**
 * Retrieves all purchase orders mapped to the authenticated supplier's brands.
 *
 * @returns {Promise<Array>} array of PurchaseOrderDto
 */
export const getPurchaseOrders = () => 
  api.get('/api/supplier/purchase-orders').then((res) => res.data);

/**
 * Updates the shipment status and estimated delivery timeline of a purchase order.
 *
 * @param {number} orderId the ID of the purchase order
 * @param {Object} updateData payload containing status and estimated delivery timeline
 * @returns {Promise<Object>} the updated purchase order
 */
export const updatePurchaseOrderStatus = (orderId, updateData) => 
  api.patch(`/api/supplier/purchase-orders/${orderId}/status`, updateData).then((res) => res.data);

File: backend/src/main/java/com/urbanfresh/service/SupplierPurchaseOrderService.java
package com.urbanfresh.service;

import java.util.List;

import com.urbanfresh.dto.request.UpdatePurchaseOrderStatusDto;
import com.urbanfresh.dto.response.PurchaseOrderDto;

/**
 * Service Layer - Manages supplier-facing purchase order operations.
 */
public interface SupplierPurchaseOrderService {

    /**
     * Retrieve all purchase orders mapped to the brands of a specific supplier.
     * 
     * @param supplierId the ID of the authenticated supplier user.
     * @return a list of DTOs representing the scoped purchase orders.
     */
    List<PurchaseOrderDto> getPurchaseOrdersForSupplier(Long supplierId);

    /**
     * Update the shipment status and estimated delivery timeline of a purchase order.
     * Enforces brand-scoping (403 Forbidden if accessed incorrectly).
     * 
     * @param supplierId the ID of the authenticated supplier user.
     * @param orderId the ID of the purchase order.
     * @param updateDto the status and delivery timeline updates.
     * @return the updated purchase order DTO.
     */
    PurchaseOrderDto updateShipmentStatus(Long supplierId, Long orderId, UpdatePurchaseOrderStatusDto updateDto);
}